### PR TITLE
ZB V2 Overhaul to Typescript

### PR DESCRIPTION
**Pull request by justin-reeves**
This completely overhauls the primary Zerobase.io site to utilize Typescript. 

Changes page layouts and reduces package usage as much as possible.

### CODE REVIEW

File: src/App.test.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import App from './App';
+
+test('renders learn react link', () => {
+  render(<App />);
+  const linkElement = screen.getByText(/learn react/i);
+  expect(linkElement).toBeInTheDocument();
+});
```
Comment (toadzky) [inline src/App.test.tsx:9]: this test should fail since we shouldn't have that string in our website.

File: src/pages/about/components/Faq.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import { Link } from "react-router-dom";
+
+/* Bootstrap */
+import Col from "react-bootstrap/Col";
+import ListGroup from "react-bootstrap/ListGroup";
+import Row from "react-bootstrap/Row";
+import Tab from "react-bootstrap/Tab";
+
+const Faq = () => (
+  <>
```
Comment (toadzky) [inline src/pages/about/components/Faq.tsx:11]: why is there an empty tag here?

File: src/pages/policies/privacy/components/Commitment.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import Row from "react-bootstrap/Row";
+
+/* Local Components */
+import SitePage from "@/components/Site/SitePage/SitePage";
+
+/* Images */
+import HeroImage from "@/images/zb-hero.jpg";
+
+const Commitment = () => (
+  <>
```
Comment (toadzky) [inline src/pages/policies/privacy/components/Commitment.tsx:14]: another empty tag to start things off

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{
+  "extends": "./tsconfig.extend.json",
+  "compilerOptions": {
+    "baseUrl": ".",
+    "target": "es5",
```
Comment (toadzky) [inline tsconfig.json:5]: why are we targeting ES5? ES6 is like 5-6 years old at this point and is supported by basically everything. is there a specific reason to target ES5?

**Review by toadzky — COMMENTED** [review]
looks fine for the most part. a couple comments, and also a question: why do you have everything with such a short line width? it looks like you have this abiding by the extremely old 80 characters. there's not really a good reason to do it especially given how indented everything is, so it just makes files a lot longer than they need to be. 120 or 132 is usually a better wrap point IME

**Comment by toadzky** [comment]
one other question: have you shown the new stuff to aron? i'm not sure if it looks the same or not, but with a change this big, i'd like to have him look it over first and give it a thumbs up on here.

File: src/App.test.tsx (hunk 1 above)
Comment (justin-reeves) [reply src/App.test.tsx:9]: forgot to delete this file. Will be writing tests later

File: src/pages/policies/privacy/components/Commitment.tsx (hunk 3 above)
Comment (justin-reeves) [reply src/pages/policies/privacy/components/Commitment.tsx:14]: Another fragment component

File: tsconfig.json (hunk 4 above)
Comment (justin-reeves) [reply tsconfig.json:5]: this was the default value from Create React App. I just forgot to change it! Woops

**Review by toadzky — APPROVED** [review]

**Pull request merged**